src/Modules/ExteriorPowers/FreeModules.jl

```diff
@@ -9,42 +9,44 @@
 
 # User facing constructor for ⋀ ᵖ F.
 function exterior_power(F::FreeMod, p::Int; cached::Bool=true)
-  (p < 0 || p > rank(F)) && error("index out of bounds")
+  @req 0 <= p <= rank(F) "Exponent out of bounds"
 
   if cached
     powers = _exterior_powers(F)
-    haskey(powers, p) && return powers[p]::Tuple{typeof(F), <:Map}
+    haskey(powers, p) && return powers[p]
   end
 
-  R = base_ring(F)
+  R = base_ring(F)::base_ring_type(F)
   n = rank(F)
-  result = FreeMod(R, binomial(n, p))
+  result_ = free_module(R, binomial(n, p))
```

Comment:
I'm not sure that this is the right thing to do. `free_module` has not been completely compatible with `FreeMod` and this means that you might get the wrong type (`FPModule` rather than `ModuleFP`). Are you sure this is correct here?

Reply:
I didn't know about that
```suggestion
  result_ = FreeMod(R, binomial(n, p))
```